service/test/BatchServerTest.cpp

```diff
@@ -426,3 +730,441 @@ TEST_F(BatchServerTest, fork_with_setup)
+    /// It takes the write_pipe_fd and the PID of the parent process, which is the client.
+    std::function<void(int, int)> child_process_func = [this](int write_pipe_fd, int client_pid)
+    {
+        // Create new BatchServer object
+        std::shared_ptr<BatchServerImp> batch_server_test = std::make_shared<BatchServerImp>(
+            client_pid,
+            m_signal_config,
+            m_control_config,
+            *m_pio_ptr,
+            m_batch_status,
+            nullptr,           // Create a real POSIXSignal because we want to use sig_action()
+            m_signal_shmem,
+            m_control_shmem,
+            m_server_pid);
+
+        // There is no EXPECT_CALL for m_posix_signal->make_sigset() and m_posix_signal->sig_action()
+        // because we are using the real POSIXSignal instead of the mock object.
+
+        // register the action_sigterm()
+        // This uses the real POSIXSignal instead of the mock object.
+        batch_server_test->child_register_handler();
+
+        int idx = 0;
+        for (const auto &request : m_signal_config) {
+        EXPECT_CALL(*m_pio_ptr, push_signal(request.name, request.domain,
+                                            request.domain_idx))
+            .WillOnce(Return(idx))
+            .RetiresOnSaturation();
+        ++idx;
+        }
+
+        idx = 0;
+        for (const auto &request : m_control_config) {
+        EXPECT_CALL(*m_pio_ptr, push_control(request.name, request.domain,
+                                             request.domain_idx))
+            .WillOnce(Return(idx))
+            .RetiresOnSaturation();
+        ++idx;
+        }
+
+        // m_batch_status->receive_message() is called in BatchServerImp::read_message()
+        // Internally it is blocking on the read() call, waiting on a message from the client.
+        // Then the client sends it a SIGTERM, which activates action_sigterm()
+        // and receive_message() throws an exception in the check_return().
+        // This exception is caught in the BatchServerImp::read_message()
+        // and char in_message = BatchStatus::M_MESSAGE_TERMINATE is returned to BatchServerImp::event_loop()
+        EXPECT_CALL(*m_batch_status, receive_message())
+        .WillOnce(
+            [&write_pipe_fd](){
+                /* Extra code for synchronizing the server. */
+               char unique_char = '!';
+               write(write_pipe_fd, &unique_char, sizeof(unique_char));
+
+                sleep(1024);
+                throw geopm::Exception("BatchStatusImp: System call failed: " "read(2)",
+                                       EINTR, __FILE__, __LINE__);
+
+                // This is irrelevant, it doesn't get returned because of the throw statement.
+                // This is just to match the return type of BatchStatusImp::receive_message()
+                return '$';
+            }
+        )
+        .RetiresOnSaturation();
```

Comment:
```suggestion
        EXPECT_CALL(*m_batch_status, receive_message())
            .WillOnce([&write_pipe_fd]() {
                /* Extra code for synchronizing the server. */
                char unique_char = '!';
                write(write_pipe_fd, &unique_char, sizeof(unique_char));
                sleep(1024);
                throw geopm::Exception("BatchStatusImp: System call failed: " "read(2)",
                                       EINTR, __FILE__, __LINE__);
                // This is irrelevant, it doesn't get returned because of the throw statement.
                // This is just to match the return type of BatchStatusImp::receive_message()
                return '$';
            }).RetiresOnSaturation();
```

For any lambdas that cannot be converted to ```Throw``` actions, please follow this style.  Here's an example from another test:
https://github.com/geopm/geopm/blob/07334f10777d04914179bb8d8649f3b6d6f92488/service/test/SSTIOTest.cpp#L229-L233